apps/dashr-lastodash/app.R

```diff
@@ -0,0 +1,379 @@
+library(readr)
+library(stringr)
+library(gdata)
+library(tidyr)
+library(dplyr)
+library(glue)
+library(ggplot2)
+library(dash)
+library(dashHtmlComponents)
+library(dashCoreComponents)
+library(dashTable)
+library(data.table)
+library(plotly)
+
+####################################################################################################
+# INITIATE APP
+
+appName <- Sys.getenv("DASH_APP_NAME")
+if (appName != ""){
+  pathPrefix <- sprintf("/%s/", appName)
+  Sys.setenv(
+    DASH_ROUTES_PATHNAME_PREFIX = pathPrefix,
+    DASH_REQUESTS_PATHNAME_PREFIX = pathPrefix
+  )
+  setwd(sprintf("/app/apps/%s", appName))
+}
+
+app <- Dash$new(name = 'DashR LAS Report')
+
+####################################################################################################
+# IMPORT DATA
+
+las_dir <- "data/"
+las_file <- "alcor2.las"
+las_path <- paste(las_dir, las_file, sep = "")
+
+####################################################################################################
+# HANDLE & STORE DATA
+
+las_object <- setRefClass("las_object", fields = list(
+  V = "data.frame", 
+  W = "data.frame", 
+  C = "data.frame", 
+  A = "data.frame")
+)
+
+convert_las <- function(las_path) {
+  las_data <- las_object()
+  las_string <- read_file(las_path)
+  las_sections <- str_split(las_string, "~")
+  for (section in las_sections[[1]]) {
+    # store section containing version and wrap mode information
+    if (startsWith(section, "V")) {
```

Comment:
@sacul-git I think this is why @ohaanika needs `gdata`; the package contains the `startsWith` function, which is Python-esque.

One way to rewrite this to eliminate the call to `startsWith` might be to try instead

```
> x <- "Test"
> if (substring(x, 1, 1) == "T") print("Found it!")
[1] "Found it!"
```

Reply:
That makes sense. `startsWith` was also added to [base in `3.3`](https://stat.ethz.ch/R-manual/R-devel/library/base/html/startsWith.html), not sure if the syntax is the same between `base` and `gdata` though.

Reply:
Ah nice that's why it worked without `gdata`! Since they say `startsWith` is faster, I'll stick to `startsWith` but remove the `library(gdata)`.